### ISSUE TITLE
customBodyRender bug

### ISSUE THREAD

**Issue by Lilliance** [gregnb/mui-datatables#915]
Hello!

When i trying to provide `customBodyRender` function in column options its broke filter with that column.

Errors heading looks like:
1. "Failed prop type: Invalid prop `primary` supplied to `ForwardRef(ListItemText)`, expected a ReactNode."
2. "Failed prop type: Invalid prop `children` supplied to `ForwardRef(Typography)`, expected a ReactNode."
3. "Invariant Violation: Objects are not valid as a React child (found: object with keys {id, name}). If you meant to render a collection of children, use an array instead."

I trying to use this `customBodyRender` function:
```
(field: { id: number; name: string }) => {
  return (
    <span style={{ color: STATUS_MAP[field.name] || 'inherit' }}>
      {field.name}
    </span>
  );
};
```

When i trying to return only `field.name` string, without `<span></span>` wrapper all works fine.

Also, your [example](https://github.com/gregnb/mui-datatables/blob/master/examples/component/index.js) works.

What i missed?

| Tech         | Version |
|--------------|---------|
| Material-UI  |    4.3.3     |
| MUI-datatables  |    2.1.8     |
| React        |    16.8.6     |
| browser      |    any     |

**Comment by gabrielliwerant** [comment]
Hello @Lilliance. You list your version of this library here as `2.1.8`, is that correct? Can you confirm that the issue still exists in the latest version, which is `2.10.3`?

**Comment by Lilliance** [comment]
Thank you for advice, @gabrielliwerant. I updated library to `2.10.3`, but i have same issue. :(

**Comment by gabrielliwerant** [comment]
It looks like be using incorrect arguments in the function. Check out the `component` example in the library. The arguments are `value`,, `tableMeta`, `updateValue`. `value` is a string, and it is the value intended for the cells in the column you are customizing.

**Comment by Lilliance** [comment]
When i don't use any args, i have same issue. Like this:
```
customBodyRender: () => {
  return (
    <span>
      hello! I am hardcoded text. 
    </span>
  );
};
```

**Comment by gabrielliwerant** [comment]
I'll need to see an example of your problem with all the (minimal) code necessary to create the issue. Can you create a codesandbox (https://codesandbox.io) and share it here?

**Comment by Lilliance** [comment]
Sure, you can check my [example](https://codesandbox.io/embed/heuristic-brook-rzp7d)
I noticed one more thing:
if you set `filterType` to `dropdown` - problem solves, but in `multiselect` or `checkbox` cases still exists.

**Comment by gabrielliwerant** [comment]
Thank you! I can reproduce the problem now. Will take a look.

**Comment by gabrielliwerant** [comment]
Ok, I've got a better handle on this now. I'm downgrading the issue to an "enhancement".

Fundamentally, the table isn't set up to use objects the way you are attempting, so the issue is deeper than filters or `customBodyRender`. Columns have to be set up to work with single values like strings or numbers. If your data has other objects in it, then you'll need to tell the column which data you want to use in the name field (e.g. `name: "field.id"`, also see the `data-as-objects` example).

So the problem with the table is that it's allowing the object to propagate through the code until it blows up. I'm working on PR that I'll mention here when pushed, which will throw an error earlier in the execution of the code so that you and others can see it and deal with it before it becomes some generic react error that's difficult to trace.

**Comment by Lilliance** [comment]
@gabrielliwerant, thank you for explanation, i appreciate your help! 
I thought about that, but in some cases if i use cell data as object all work fine. 
But now i see it was just non-documented luck :), ty.

**Comment by gabrielliwerant** [comment]
@Lilliance That's correct, it looks like you got lucky! I think you would have run into problems eventually. Not your fault, as there were no error messages to help out here, but hopefully the additions will provide something good enough for now.
